src/main/java/com/example/gradecalculator/controller/UserController.java

```diff
@@ -65,17 +68,22 @@ public String myProfileGet(Model model, Authentication authentication){
 
     @GetMapping("/editProfile")
     public String editProfileGet(Model model, Authentication authentication){
-        var userTypes = userTypeRepository.findAll();
-        model.addAttribute("userTypes", userTypes);
-
-        var userID = userService.getAuthenticatedUserId(authentication);
-        var user = userRepository.findById(userID);
-        var userData = userMapper.dataToTO(user.get());
+        var userAuthenticated = userService.getAuthenticatedUserId(authentication);
+        var userID = userRepository.findById(userAuthenticated);
+        var userData = userMapper.dataToTO(userID.get());
         model.addAttribute("editProfile", userData);
 
         return "editProfile";
     }
 
+    @GetMapping("/editPassword")
+    public String editPasswordGet(Model model, Authentication authentication){
+        var userAuthenticated = userService.getAuthenticatedUserId(authentication);
+        var userID = userRepository.findById(userAuthenticated);
+        var userData = userMapper.dataToTO(userID.get());
```

Comment:
1. I do not think you use this data, its not getting added to the model
2. duplicated code from editProfile -> refactor into a method that gets reused